### ISSUE TITLE
App crashes when loading comments by users with image flairs

### ISSUE THREAD

**Issue by aidankay1** [ccrama/Slide#3292]
Slide version: 6.6
Android version: 10 (OxygenOS 10.3.6)

Opening any post on r/Splatoon or r/TF2 that has comments causes the app to crash, though previewing images and videos works fine. Both of these subreddits have user flairs that contain images. Subreddits that use placeholders in the flair text (e.g. :image:) are not affected.

Selecting "sync subreddit image flairs" from the sidebar temporarily fixes the issue. When the app is closed and opened again, the issue reappears on the affected subreddits. Occasionally, when reopening the app after a crash, it will launch with certain elements of the default theme (dark gray/red) but will still mostly use my custom theme.

**Comment by tp0** [comment]
Was also reported on Slide's subreddit:

https://reddit.com/r/slideforreddit/comments/jt05sh/v66_crashes_on_some_subreddits_eg_randroid_but/

**Comment by tp0** [comment]
> When the app is closed and opened again, the issue reappears on the affected subreddits.

I can still exit the app (back button > "exit?" dialog) and I don't have to do the resync, but if remove the app from the recent apps view or force kill it then I have to (something to do with memory handling?).

**Comment by Sobsz** [comment]
here's an odd one: a child comment of [this](https://reddit.com/r/softwaregore/comments/jyixf8/speechnt/gd4hcfp?context=3) (probably the beflaired one) crashes the app but syncing reports no image flairs

**Comment by HeyBanditoz** [comment]
You guys get a stacktrace similar to this?
<details>
    <summary>Stacktrace</summary>

```
FATAL EXCEPTION: main
Process: me.ccrama.redditslide, PID: 15033
java.lang.IllegalStateException: ImageLoader must be init with configuration before using
	at com.nostra13.universalimageloader.core.ImageLoader.checkConfiguration(ImageLoader.java:613)
	at com.nostra13.universalimageloader.core.ImageLoader.getDiskCache(ImageLoader.java:659)
	at me.ccrama.redditslide.Adapters.CommentAdapter.onBindViewHolder(CommentAdapter.java:416)
	at androidx.recyclerview.widget.RecyclerView$Adapter.onBindViewHolder(RecyclerView.java:7065)
	at androidx.recyclerview.widget.RecyclerView$Adapter.bindViewHolder(RecyclerView.java:7107)
	at androidx.recyclerview.widget.RecyclerView$Recycler.tryBindViewHolderByDeadline(RecyclerView.java:6012)
	at androidx.recyclerview.widget.RecyclerView$Recycler.tryGetViewHolderForPositionByDeadline(RecyclerView.java:6279)
	at androidx.recyclerview.widget.RecyclerView$Recycler.getViewForPosition(RecyclerView.java:6118)
	at androidx.recyclerview.widget.RecyclerView$Recycler.getViewForPosition(RecyclerView.java:6114)
	at androidx.recyclerview.widget.LinearLayoutManager$LayoutState.next(LinearLayoutManager.java:2303)
	at androidx.recyclerview.widget.LinearLayoutManager.layoutChunk(LinearLayoutManager.java:1627)
	at androidx.recyclerview.widget.LinearLayoutManager.fill(LinearLayoutManager.java:1587)
	at androidx.recyclerview.widget.LinearLayoutManager.onLayoutChildren(LinearLayoutManager.java:665)
	at me.ccrama.redditslide.Views.PreCachingLayoutManagerComments.onLayoutChildren(PreCachingLayoutManagerComments.java:20)
	at androidx.recyclerview.widget.RecyclerView.dispatchLayoutStep2(RecyclerView.java:4134)
	at androidx.recyclerview.widget.RecyclerView.dispatchLayout(RecyclerView.java:3851)
	at androidx.recyclerview.widget.RecyclerView.onLayout(RecyclerView.java:4404)
	at android.view.View.layout(View.java:20712)
	at android.view.ViewGroup.layout(ViewGroup.java:6194)
	at androidx.swiperefreshlayout.widget.SwipeRefreshLayout.onLayout(SwipeRefreshLayout.java:625)
	at android.view.View.layout(View.java:20712)
	at android.view.ViewGroup.layout(ViewGroup.java:6194)
	at android.widget.FrameLayout.layoutChildren(FrameLayout.java:323)
	at android.widget.FrameLayout.onLayout(FrameLayout.java:261)
	at android.view.View.layout(View.java:20712)
	at android.view.ViewGroup.layout(ViewGroup.java:6194)
	at androidx.viewpager.widget.ViewPager.onLayout(ViewPager.java:1775)
	at android.view.View.layout(View.java:20712)
	at android.view.ViewGroup.layout(ViewGroup.java:6194)
	at androidx.coordinatorlayout.widget.CoordinatorLayout.layoutChild(CoordinatorLayout.java:1213)
	at androidx.coordinatorlayout.widget.CoordinatorLayout.onLayoutChild(CoordinatorLayout.java:899)
	at androidx.coordinatorlayout.widget.CoordinatorLayout.onLayout(CoordinatorLayout.java:919)
	at android.view.View.layout(View.java:20712)
	at android.view.ViewGroup.layout(ViewGroup.java:6194)
	at android.widget.FrameLayout.layoutChildren(FrameLayout.java:323)
	at android.widget.FrameLayout.onLayout(FrameLayout.java:261)
	at android.view.View.layout(View.java:20712)
	at android.view.ViewGroup.layout(ViewGroup.java:6194)
	at android.widget.LinearLayout.setChildFrame(LinearLayout.java:1812)
	at android.widget.LinearLayout.layoutVertical(LinearLayout.java:1656)
	at android.widget.LinearLayout.onLayout(LinearLayout.java:1565)
	at android.view.View.layout(View.java:20712)
	at android.view.ViewGroup.layout(ViewGroup.java:6194)
	at android.widget.FrameLayout.layoutChildren(FrameLayout.java:323)
	at android.widget.FrameLayout.onLayout(FrameLayout.java:261)
	at android.view.View.layout(View.java:20712)
	at android.view.ViewGroup.layout(ViewGroup.java:6194)
	at android.widget.LinearLayout.setChildFrame(LinearLayout.java:1812)
	at android.widget.LinearLayout.layoutVertical(LinearLayout.java:1656)
	at android.widget.LinearLayout.onLayout(LinearLayout.java:1565)
	at android.view.View.layout(View.java:20712)
	at android.view.ViewGroup.layout(ViewGroup.java:6194)
	at me.ccrama.redditslide.SwipeLayout.SwipeBackLayout.onLayout(SwipeBackLayout.java:385)
	at android.view.View.layout(View.java:20712)
	at android.view.ViewGroup.layout(ViewGroup.java:6194)
	at android.widget.FrameLayout.layoutChildren(FrameLayout.java:323)
	at android.widget.FrameLayout.onLayout(FrameLayout.java:261)
	at com.android.internal.policy.DecorView.onLayout(DecorView.java:753)
	at android.view.View.layout(View.java:20712)
	at android.view.ViewGroup.layout(ViewGroup.java:6194)
	at android.view.ViewRootImpl.performLayout(ViewRootImpl.java:2817)
	at android.view.ViewRootImpl.performTraversals(ViewRootImpl.java:2334)
	at android.view.ViewRootImpl.doTraversal(ViewRootImpl.java:1481)
	at android.view.ViewRootImpl$TraversalRunnable.run(ViewRootImpl.java:7369)
	at android.view.Choreographer$CallbackRecord.run(Choreographer.java:1012)
	at android.view.Choreographer.doCallbacks(Choreographer.java:823)
	at android.view.Choreographer.doFrame(Choreographer.java:758)
	at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:998)
	at android.os.Handler.handleCallback(Handler.java:873)
	at android.os.Handler.dispatchMessage(Handler.java:99)
	at android.os.Looper.loop(Looper.java:193)
	at android.app.ActivityThread.main(ActivityThread.java:6671)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:860)
```
</details>

**Comment by djhopkins2** [comment]
I'm getting the same bug in version 66 as well period as soon as I got the update to this version, that became nearly unusable given that a number of the main subreddits use image flair. Quite often when I open a post or scroll down a small distance the app will completely crash and exit. The stack Trace I get is identical to the comment above. I'm on mobile right now otherwise I would paste it in here.

This is a pretty nasty bug in that it makes slide somewhat unusable.

I started following the commits for version 66 and I can see where changes were made on line 416 of commentadapter.java
